public/app/features/serviceaccounts/state/actions.ts

```diff
@@ -1,9 +1,30 @@
-import { ThunkResult } from '../../../types';
+import { ServiceAccountDTO, ThunkResult } from '../../../types';
 import { getBackendSrv } from '@grafana/runtime';
-import { serviceAccountLoaded, serviceAccountsLoaded, serviceAccountTokensLoaded } from './reducers';
+import {
+  acOptionsLoaded,
+  builtInRolesLoaded,
+  serviceAccountLoaded,
+  serviceAccountsLoaded,
+  serviceAccountTokensLoaded,
+} from './reducers';
+import { accessControlQueryParam } from 'app/core/utils/accessControl';
+import { fetchBuiltinRoles, fetchRoleOptions } from 'app/core/components/RolePicker/api';
 
 const BASE_URL = `/api/serviceaccounts`;
 
+export function fetchACOptions(): ThunkResult<void> {
+  return async (dispatch) => {
+    try {
+      const options = await fetchRoleOptions();
+      const builtInRoles = await fetchBuiltinRoles();
+      dispatch(acOptionsLoaded(options));
```

Comment:
Not sure if it makes any difference here but maybe we'd dispatch the options loaded already after they were fetch? Currently we're also waiting for the roles to be fetched.

Reply:
It should make any difference in this case,  but I'll change it so it "makes sense"